staging/src/k8s.io/apimachinery/pkg/util/sets/byte.go

```diff
@@ -52,13 +52,25 @@ func (s Byte) Insert(items ...byte) {
 	}
 }
 
+// WithInsert adds items to the set, then returns the set.
```

Comment:
Generally as a user I have a very strong expectation that a "WithX" function doesn't modify the original.

Since the existing Insert and Delete methods don't currently return anything, you can add a return value and mostly not break any clients; I'd recommend that.